### ISSUE TITLE
Ethernet/Processor Warm-up Before a Test

### ISSUE THREAD

**Issue by electroflame** [alexjustesen/speedtest-tracker#1357]
**Is your feature request related to a problem? Please describe.**
I've noticed that because this doesn't take much processing power, it allows Intel's (kind of overzealous) sleep states to be activated on either the NIC/Ethernet port, or the processor, or both (even when this is running a test).  A side effect of that is that when the speedtest is run it often won't do enough processing to trigger the "wake up" from the C-state immediately, resulting in massively reduced speeds in the results (i.e. 10s instead of 100s or 1000s).  If I instead queue two speedtests manually in a row the second one usually will have the correct speed, as whatever controller was sleeping should be awake by then.

**Describe the solution you'd like**
Something to "pre-warm" the Ethernet connection and/or the processor seems like it'd solve this.  Basically a brief job like a more basic, faster speedtest (download only, maybe?) or something similar to try and make sure all the controllers are warmed up before starting.  As a bonus, this might solve the ongoing issues with EEE (Energy-Efficient Ethernet) at the same time, as it might force the NIC controller to wake up.

**Additional context**
This was super annoying to debug, but I eventually tracked it down to something related to C-states, as disabling them in the BIOS solves it completely.   For reference, I initially had this container running on a more busy node, but after shuffling some things around this landed on an almost-empty one.  The side-effect of that is that the server's not doing enough work regularly enough to keep the processor (or NIC, or both) warmed up, so almost every speedtest fails with a super-low, incorrect speed.  If I disable C-states, or cram some more processing on the node, this issue goes away.

Also, for clarity, EEE is disabled on everything (node, switches, router, etc.) so it's not that.

**Comment by alexjustesen** [comment]
Attaching to #1354 as it's the same premise.
